### PR TITLE
fix: Exception handling for creating default migration.

### PR DESCRIPTION
**Pull request by SandPod**
### Changes
- Adds documentation about all exceptions creating a migration might throw.
- Adds better error messaging if creating a migration fails when creating a new project.
- Includes the migration creation process as part of the success reporting for creating a new project.

### After review feedback:
Screenshot 2023-12-07 at 13 53 12

## Pre-launch Checklist

- [ ] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [ ] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [ ] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

_If you have done any breaking changes, make sure to outline them here, so that they can be included in the notes for the next release._

### CODE REVIEW

File: tools/serverpod_cli/lib/src/create/database_setup.dart [hunk 1]
```diff
@@ -23,12 +24,38 @@ class DatabaseSetup {
...
+    try {
+      migration = await generator.createMigration(
+        force: false,
+        config: config,
+      );
+    } on MigrationVersionLoadException catch (e) {
+      log.error(
+        'Unable to determine latest database definition due to a corrupted '
+        'migration. Please re-create or remove the migration version and try '
+        'again. Migration version: "${e.versionName}".',
```
Comment (Isakdl) [inline tools/serverpod_cli/lib/src/create/database_setup.dart:37]: Question: In what scenario would it help to rerun?

File: tools/serverpod_cli/lib/src/create/database_setup.dart [hunk 2]
```diff
@@ -23,12 +24,38 @@ class DatabaseSetup {
...
+      );
+      log.error(e.exception);
+    } on GenerateMigrationDatabaseDefinitionException {
+      log.error('Unable to generate database definition for project.');
```
Comment (Isakdl) [inline tools/serverpod_cli/lib/src/create/database_setup.dart:41]: Question: Do we know more about what went wrong? If I would encounter this error it wouldn't really help me moving forward.

File: tools/serverpod_cli/lib/src/create/database_setup.dart [hunk 3]
```diff
@@ -23,12 +24,38 @@ class DatabaseSetup {
...
+        'Unable to create migration. A directory with the same name already '
+        'exists: "${e.directoryPath}".',
+      );
+    }
+
+    if (migration == null) {
+      log.warning(
+        'An error occurred while creating the initial migration. You might '
+        'not be set up correctly for the created project. Please see the '
+        'documentation for how to create and apply a migration manually.',
```
Comment (Isakdl) [inline tools/serverpod_cli/lib/src/create/database_setup.dart:53]: Suggestion: Perhaps add the url to the docs here.

**Review by Isakdl — COMMENTED** [review]
Some questions, see comments.

File: tools/serverpod_cli/lib/src/create/database_setup.dart (hunk 1 above)
Comment (SandPod) [reply tools/serverpod_cli/lib/src/create/database_setup.dart:37]: Basically all of these should never happen when creating a migration. Perhaps a single catch all message is better, maybe the one provided bellow with a link to the documentation.

**Review by Isakdl — APPROVED** [review]
LGTM

**Pull request merged**